src/test/ui/deriving/deriving-default-enum.rs

```diff
@@ -12,6 +12,16 @@ enum Foo {
     Beta(NotDefault),
 }
 
+// #[default] on a generic enum does not add `Default` bounds to the type params.
+#[derive(Default)]
+enum MyOption<T> {
+    #[default]
```

Comment:
I think it's a good idea to update src/test/ui/macros/macros-nonfatal-errors.rs to make sure that `#[non_exhaustive] #[default]` (currently an error, it seems) when supported does generate the right bound. I think given the non-fatal there it's possible that we're emitting code, just later erroring?

Reply:
So, to clarify, you'd like me to add something akin to:

https://github.com/rust-lang/rust/blob/5b4bd154de6e41d4739914f446349f8201ec3ef9/src/test/ui/macros/macros-nonfatal-errors.rs#L89-L95

but with a, say, `<T>`-generic type, and then assert that `Enum<NotDefault>` does not implement `Default`?

Reply:
@Mark-Simulacrum In theory it should be OK to explicitly mark a non_exhaustive variant as Default, as long as you're committing in your API to making it stay that way in the future. But I do think that should be a separate decision, and not part of this PR. So, this seems fine.